c7n/resources/aws.py

```diff
@@ -536,11 +588,21 @@ def __init__(self, ctx, config):
         super().__init__(ctx, config)
         # can't use a local session as we dont want an unassumed session cached.
         s3_client = self.ctx.session_factory(assume=False).client('s3')
-        region = s3_client.get_bucket_location(Bucket=self.bucket)['LocationConstraint']
-        # if region is None, we use us-east-1
-        region = region or "us-east-1"
+
+        # Try determining the output bucket region via HTTP requests since
+        # that works more consistently in cross-region scenarios. Fall back
+        # the GetBucketLocation API if necessary.
+        try:
+            self.bucket_region = (
+                get_bucket_region_clientless(self.bucket, s3_client._endpoint.host) or
+                get_bucket_region(self.bucket, s3_client)
+            )
+        except ClientError as err:
+            log.error('unable to determine a region for output bucket %s: %s',
+                self.bucket, err)
+            sys.exit(1)
```

Comment:
sys exit here feels odd, ie library causing programing exit, versus raised exception

Reply:
Fair, I switched that out because the exception looked unnecessarily noisy and I saw the exit pattern in a couple other spots. Exception does make more sense though, will switch back.